signature.go

```diff
@@ -48,7 +48,12 @@ func getDownloadURL(format Format, htmlPlayerFile string) (*url.URL, error) {
 	query := u.Query()
 	query.Set("ratebypass", "yes")
 	if len(sig) > 0 {
-		query.Set("signature", sig)
+		sigParam := "signature"
+		if v, ok := format.meta["sp"].(string); ok && v != "" {
+			sigParam = v
+		}
+
```

Comment:
It's a nit-pick, but @rylio only rarely uses blank lines within functions. I feel like it doesn't belong here. Do with this comment as you please.